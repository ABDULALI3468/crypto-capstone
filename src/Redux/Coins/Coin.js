// import { useParams } from "react-router-dom";
const FETCH_COIN = "FETCH_COIN";

export const coinFetchAction = (coin) => ({
  type: FETCH_COIN,
  coin,
});


// const params = useParams();
export const fetchCoin = (url) => {
  // Thunk Function
  return async (dispatch, getState) => {
    // Fetching results from an API : asynchronous action
    const response = await fetch(url);
    const data = await response.json();
    console.log(data);

    // Dispatching the action when async
    // action has completed.
    dispatch(coinFetchAction(data));
  };
};

const coin = (state = [], action) => {
  switch (action.type) {
    case FETCH_COIN:
      return action.coin;
    default:
      return state;
  }
};

export default coin;
