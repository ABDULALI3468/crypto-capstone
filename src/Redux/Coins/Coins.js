const FETCH_COINS = 'FETCH_COINS';

export const coinsFetchAction = (coins) => ({
  type: FETCH_COINS,
  coins,
});

export const fetchCoins = () => async (dispatch) => {
  const response = await fetch('https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=50&page=1&sparkline=false');
  const data = await response.json();
  dispatch(coinsFetchAction(data));
};

const coins = (state = [], action) => {
  switch (action.type) {
    case FETCH_COINS:
      return action.coins;
    default:
      return state;
  }
};

export default coins;
