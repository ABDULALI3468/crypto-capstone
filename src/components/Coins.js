import { useEffect, React } from "react";
import { useDispatch, useSelector } from "react-redux";
import { fetchCoins } from "../Redux/Coins/Coins";
import { Link } from "react-router-dom";
import Coin from "./Coin";
import "../styles/coins.css";
import mainImage from "../styles/crypto-main.png";

const Coins = () => {
  const dispatch = useDispatch();
  const coins = useSelector((state) => state.coins);

  useEffect(() => {
    if (coins.length === 0) {
      dispatch(fetchCoins());
    }
  }, [coins.length]);

  return (
    <>
      <div className="container">
        <div>
          <img className="mainImage" src={mainImage} />
        </div>
        <h1 className="neon">Coins Overview</h1>
        <h1 className="neon">Coins Overview</h1>
        <h1 className="neon">Coins Overview</h1>
        <div className="coinsContainer">
          {coins.map((coin) => {
            return (
              <Link className="link" to={`/coin/${coin.id}`} element={<Coin />} key={coin.id}>
                <div className="coinContainer">
                  <div className="list">
                    <img className="coinImage" src={coin.image} alt="coinImage" />
                    <p className="coinId">{`${coin.symbol.charAt(0).toUpperCase() + coin.symbol.slice(1)}/USD`}</p>
                  </div>
                  <p className="currentPrice">AT: ${coin.current_price.toFixed(1)}</p>
                </div>
              </Link>
            );
          })}
        </div>
      </div>
    </>
  );
};

export default Coins;

// {
//     "id": "bitcoin",
//     "symbol": "btc",
//     "name": "Bitcoin",
//     "image": "https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579",
//     "current_price": 22812,
//     "market_cap": 435759766618,
//     "market_cap_rank": 1,
//     "fully_diluted_valuation": 479055878836,
//     "total_volume": 39190406503,
//     "high_24h": 22813,
//     "low_24h": 22048,
//     "price_change_24h": 611.89,
//     "price_change_percentage_24h": 2.75623,
//     "market_cap_change_24h": 11687795171,
//     "market_cap_change_percentage_24h": 2.75609,
//     "circulating_supply": 19102062,
//     "total_supply": 21000000,
//     "max_supply": 21000000,
//     "ath": 69045,
//     "ath_change_percentage": -66.97312,
//     "ath_date": "2021-11-10T14:24:11.849Z",
//     "atl": 67.81,
//     "atl_change_percentage": 33528.7659,
//     "atl_date": "2013-07-06T00:00:00.000Z",
//     "roi": null,
//     "last_updated": "2022-07-24T15:49:35.677Z"
// }
