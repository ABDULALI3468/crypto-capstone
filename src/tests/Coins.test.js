import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import store from '../Redux/ConfigureStore';
import Coins from '../components/Coins';

test('Popular component testing', () => {
  const TREE = TestRenderer.create(
    <Provider store={store}>
      <Coins />
    </Provider>,
  );
  expect(TREE).toMatchSnapshot();
});
