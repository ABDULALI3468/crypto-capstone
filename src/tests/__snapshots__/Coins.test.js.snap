// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popular component testing 1`] = `
<div
  className="container"
>
  <div
    className="imageContainer"
  >
    <img
      className="mainImage"
      src="mainimage.gif"
    />
  </div>
  <h1
    className="neon"
  >
    Coins Overview
  </h1>
  <div
    className="coinsContainer"
  />
</div>
`;
